src/main/scala/com/amazon/deequ/analyzers/applicability/Applicability.scala

```diff
@@ -31,13 +31,98 @@ private[deequ] case class ApplicabilityResult(
   failures: Seq[(String, Throwable)]
 )
 
+private[deequ] object Applicability {
+  def randomString(nullable: Boolean = false): java.lang.String = {
+    if (nullable && math.random < 0.01) {
+      null
+    } else {
+      val length = (math.random * 20 + 1).toInt
+      RandomStringUtils.randomAlphanumeric(length)
+    }
+  }
+
+  def randomInteger(nullable: Boolean = false): java.lang.Integer = {
```

Comment:
we should refactor the random number generation out of all the typed methods into a single place and only invoke the conversion in the actual methods.